core/spam/statistics.go

```diff
@@ -0,0 +1,23 @@
+package spam
+
+type Statistics struct {
+	Proposals         Statistic
+	Delegations       Statistic
+	Transfers         Statistic
+	NodeAnnouncements Statistic
+	Votes             []VoteStatistic
+}
+
+type VoteStatistic struct {
```

Comment:
I'm not a fan of having these intermediary structs - what's the point of them? we only create them temporarily before we convert to the proto ones so why not use the proto ones directly? may not be super clean from a pure point of view but for something like that I'd prefer performance.